src/cpuid/src/template/mod.rs

```diff
@@ -3,3 +3,22 @@
 
 // Contains Intel specific templates.
 pub mod intel;
+
+use std::collections::HashSet;
+
+use kvm_bindings::CpuId;
+
+use crate::common::{get_vendor_id_from_cpuid, VENDOR_ID_INTEL};
+use crate::transformer::Error;
+
+/// Returns MSRs to be saved based on CPUID features that are enabled.
+pub fn msrs_to_save_by_cpuid(cpuid: &CpuId) -> Result<HashSet<u32>, Error> {
+    let vendor_id = get_vendor_id_from_cpuid(cpuid).map_err(|_| Error::InvalidVendor)?;
```

Comment:
nit: not sure what the convention is, but could be just (I guess)
```rust
.map_err(Error::InvalidVendor)
```

Reply:
It is this way because we map `cpuid::common::Error` to `cpuid::transformer::Error`.